core/src/main/scala/com/spotify/featran/transformers/NHotEncoder.scala

```diff
@@ -25,26 +25,50 @@ import scala.collection.SortedMap
 /**
  * Transform a collection of categorical features to binary columns, with at most N one-values.
  *
- * Missing values are transformed to zero vectors.
+ * Missing values are either transformed to zero vectors or encoded as a missing value.
  *
- * When using aggregated feature summary from a previous session, unseen labels are ignored and
- * [[FeatureRejection.Unseen]] rejections are reported.
+ * When using aggregated feature summary from a previous session, unseen labels are either
+ * transformed to zero vectors or encoded as a missing value (if missingValueOpt is provided) and
+ * [FeatureRejection.Unseen]] rejections are reported.
  */
 object NHotEncoder {
   /**
-   * Create a new [[NHotEncoder]] instance.
-   */
-  def apply(name: String): Transformer[Seq[String], Set[String], SortedMap[String, Int]] =
-    new NHotEncoder(name)
+    * Create a new [[NHotEncoder]] instance.
+    */
+  def apply(name: String, missingValueOpt: Option[String] = None):
+  Transformer[Seq[String], Set[String], SortedMap[String, Int]] =
+    new NHotEncoder(name, missingValueOpt)
+
+  def apply(name: String, missingValue: String):
```

Comment:
`missingValue` is the column/feature name for unseen labels right? It might give user too much freedom and be error-prone. How about making it a `encodeMissingValue: Boolean`, and we hard code a name for unseen/missing values?

For example, given `name = "onehot"` and labels `["a", "b", "c"]`, we'll get
```
onehot_a
onehot_b
onehot_c
onehot_missing // or with extra escape character to avoid collision
```